bot/helpers/db.py

```diff
@@ -0,0 +1,195 @@
+# db.py
+
+
+class DBHelper:
+    """ Class to contain database query wrapper functions. """
+
+    def __init__(self, pool):
+        """ Set attributes. """
+        self.pool = pool
+
+    @staticmethod
+    def _get_record_attrs(records, key='id'):
+        """ Get key list of attributes from list of Record objects. """
+        return list(map(lambda r: r[key], records))
+
+    async def _get_row(self, table, obj):
+        """ Generic method to get table row by object id. """
+        statement = (
+            f'SELECT * FROM {table}\n'
+            '   WHERE id = $1'
+        )
+
+        async with self.pool.acquire() as connection:
+            async with connection.transaction():
+                row = await connection.fetchrow(statement, obj.id)
+
+        return {col: val for col, val in row.items()}
+
+    async def _update_row(self, table, obj, **data):
+        """ Generic method to update table row by object id. """
+        col_vals = ',\n    '.join(f'{col} = {val}' for col, val in data.items())
+        ret_vals = ',\n    '.join(data)
+        statement = (
+            f'UPDATE {table}\n'
+            f'SET {col_vals}\n'
+            'WHERE\n'
+            '    id = $1\n'
+            f'RETURNING {ret_vals};'
+        )
+
+        async with self.pool.acquire() as connection:
+            async with connection.transaction():
+                updated_vals = await connection.fetch(statement, obj.id)
+
+        return {col: val for rec in updated_vals for col, val in rec.items()}
+
+    async def insert_guilds(self, *guilds):
+        """ Add a list of guilds into the guilds table and return the ones successfully added. """
+        rows = [(guild.id, None, None, None) for guild in guilds]
+        statement = (
+            'INSERT INTO guilds (id)\n'
+            '    (SELECT id FROM unnest($1::guilds[]))\n'
+            '    ON CONFLICT (id) DO NOTHING\n'
+            '    RETURNING id;'
```

Comment:
See above

Reply:
There's no strings formatting in this one though?

Reply:
Just assumed that `unnest($1::guilds[]))` would be unpacking values from the variable, I'm probably wrong figured it was just best to check